lib/ansible/parsing/vault/__init__.py

```diff
@@ -631,26 +665,25 @@ def gen_key_initctr(self, password, salt):
 
         key1 = derivedkey[:keylength]
         key2 = derivedkey[keylength:(keylength * 2)]
-        iv = derivedkey[(keylength * 2):(keylength * 2) + ivlength]
 
-        return key1, key2, hexlify(iv)
+        iv = 0
+        if generate_iv:
```

Comment:
why does it need an iv for backwards compatibility? Surely old clients will just read the 0 IV and work fine?